source/graphics/MapGraphic.js

```diff
@@ -34,9 +34,23 @@ export default class MapGraphic extends Graphic {
 
     // compute initial cartogram
     this.updatePreProjection()
+    const baseMapTopoJson = mapResource.getTopoJson()
+    let filteredTopoJson = null
+    let filteredGeometries = null
+
+    // for custom uploads with incomplete data
+    if (properties.length !== baseMapTopoJson.objects.states.geometries.length) {
+      const statesWithData = properties.map(property => property[0])
+      filteredGeometries = baseMapTopoJson.objects.states.geometries
+        .filter(geom => statesWithData.indexOf(geom.id) > -1)
+      filteredTopoJson = JSON.parse(JSON.stringify(baseMapTopoJson)) // clones the baseMap
+      // only pass filtered geometries to topogram generator
+      filteredTopoJson.objects.states.geometries = filteredGeometries
+    }
```

Comment:
Can we factor this filtering logic out into a helper method to keep `computeCartogram` short and high-level?